app/src/main/scala/org/alephium/explorer/persistence/queries/BlockQueries.scala

```diff
@@ -74,6 +74,16 @@ object BlockQueries extends StrictLogging {
       header <- BlockHeaderSchema.table.filter(_.hash === hash).result.headOption
     } yield header.map(_.toLiteApi)
 
+  /** For a given `BlockHash` returns its basic chain information */
+  def getBlockChainInfo(hash: BlockHash): DBActionR[Option[(GroupIndex, GroupIndex, Boolean)]] =
+    BlockHeaderSchema.table
+      .filter(_.hash === hash)
+      .map { block =>
+        (block.chainFrom, block.chainTo, block.mainChain)
+      }
+      .result
+      .headOption
```

Comment:
should we try to directly use the raw sql format instead? I know it's a bit more painful, but we should  force our-self to do it (I'm the first one to blame for not doing it)

Reply:
Ah yep. Done :)